src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -90,12 +90,26 @@ public async virtual Task<HostSecretsInfo> GetHostSecretsAsync()
                         _logger.LogDebug("Loading host secrets");
 
                         hostSecrets = await LoadSecretsAsync<HostSecrets>();
-                        if (hostSecrets == null)
+                        try
+                        {
+                            if (hostSecrets == null)
+                            {
+                                // host secrets do not yet exist so generate them
+                                _logger.LogDebug(Resources.TraceHostSecretGeneration);
+                                hostSecrets = GenerateHostSecrets();
+                                await PersistSecretsAsync(hostSecrets);
+                            }
+                        }
+                        catch (Exception ex)
```

Comment:
Is there a more specific exception you can catch?

Reply:
I was wondering this as well... but because there's a handful of different `ISecretsRepository` implementations (Blob, Kubernetes, KeyVault, FileSystem, etc), we can't really know every possibility of exception.

Reply:
I can leave this as is right now and once we have more data to validate the common exceptions thrown we can change it